src/component.js

```diff
@@ -19,27 +19,32 @@ export default {
     };
   },
   created() {
-    this.widgetFactory = this.connector(this.updateData, () => {});
-    this.widget = this.widgetFactory(this.widgetParams);
+    this.factory = this.connector(this.updateState, () => {});
+    this.widget = this.factory(this.widgetParams);
     this.instantSearchInstance.addWidget(this.widget);
   },
   beforeDestroy() {
     this.instantSearchInstance.removeWidget(this.widget);
   },
   watch: {
-    widgetParams(newVal) {
-      const oldWidget = this.widget;
-      this.widget = this.widgetFactory(newVal);
+    widgetParams(nextWidgetParams) {
+      this.state = null;
+      this.instantSearchInstance.removeWidget(this.widget);
+      this.widget = this.factory(nextWidgetParams);
       this.instantSearchInstance.addWidget(this.widget);
-      this.instantSearchInstance.removeWidget(oldWidget);
     },
   },
   methods: {
     suit(...args) {
       return suit(this.widgetName, ...args);
     },
-    updateData(state = {}) {
-      this.state = state;
+    updateState(state = {}, isFirstRender) {
+      if (!isFirstRender) {
+        // Avoid to update the state on first render
+        // otherwise we have a flash from empty state
+        // to the next state
```

Comment:
Avoid updating the state on first render, otherwise there will be a flash of placeholder data